### PR TITLE
add support for influx line protocol

### PR DESCRIPTION
**Pull request by megastef**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,33 @@ var testAgent = client.createAgent(new SpmAgent.Agent ({
 testAgent.on ('metrics', console.log)
 ```
 
+### Using InfluxDB format 
+
+Sematext Cloud supports the influx line protocol.
```
Comment (otisg) [inline README.md]: Consistent casing, see next line

File: README.md [hunk 2]
```diff
@@ -32,6 +32,33 @@ var testAgent = client.createAgent(new SpmAgent.Agent ({
...
+```js
+
+var SpmAgent = require('spm-agent')
+var client = new SpmAgent()
+var testAgent = client.createAgent(new SpmAgent.Agent ({
+  start: function (agent) {
+    // initialize your metrics collector ...
+    this.tid = setInterval(function () {
+      // get every 30 seconds some metrics 
+      // SPM gets an array of metrics for a specific app
```
Comment (otisg) [inline README.md]: What's SPM? :)

File: README.md [hunk 3]
```diff
@@ -32,6 +32,33 @@ var testAgent = client.createAgent(new SpmAgent.Agent ({
...
+
+```js
+
+var SpmAgent = require('spm-agent')
+var client = new SpmAgent()
+var testAgent = client.createAgent(new SpmAgent.Agent ({
+  start: function (agent) {
+    // initialize your metrics collector ...
+    this.tid = setInterval(function () {
+      // get every 30 seconds some metrics
```
Comment (otisg) [inline README.md]: Hm, I don't see 30 anywhere...

File: README.md [hunk 4]
```diff
@@ -32,6 +32,33 @@ var testAgent = client.createAgent(new SpmAgent.Agent ({
...
+      // SPM gets an array of metrics for a specific app
+       agent.addMetrics({
+       	measurement: 'test-app', 
+       	tags: {role: 'frontend'}, 
+       	fields: {requestCount: 1}}
+       )
+    }, client.config.collectionInterval)
+  }
+}))
+// monitor which values we added by "addMetrics"
```
Comment (otisg) [inline README.md]: monitor which => monitor whose

File: lib/index.js [hunk 5]
```diff
@@ -28,7 +28,9 @@ function SpmAgent (spmUrl) {
     retransmit: 0
   }
   var SpmSender = require('./sender/spmsender.js')
+  var InfluxDbSender = require('./sender/influxsender.js')
```
Comment (otisg) [inline lib/index.js]: InfluxDbSender vs. InfluxSender below

File: lib/index.js [hunk 6]
```diff
@@ -63,6 +70,26 @@ function SpmAgent (spmUrl) {
...
+    counters.error += 1
+    logger.error('Error in spm-sender ' + JSON.stringify(e))
+  })
+  this.influxSender.on('retransmit', function () {
+    counters.retransmit += 1
+    // logger.warn ('retransmit event ' + JSON.stringify (counters))
+    this.emit('stats', counters)
+  }.bind(this))
   if (!config.tokens.spm && !process.env.LOGSENE_STATS_TOKEN) {
     var msg = 'Please set SPM App Token e.g. export SPM_TOKEN="YOUR-SPM-APP-TOKEN"'
```
Comment (otisg) [inline lib/index.js]: SPM App Token => Monitoring App Token

File: README.md [hunk 7]
```diff
@@ -12,7 +12,7 @@
 - Configuration handling 
 - Pluggable agents
 
-Example to implement a monitoring agent:
+## Example to implement a monitoring agent:
```
Comment (otisg) [inline README.md]: For sending customer metrics?

Comment (megastef) [reply README.md]: For all metrics. There is no difference anymore.

File: README.md [hunk 8]
```diff
@@ -32,6 +32,33 @@ var testAgent = client.createAgent(new SpmAgent.Agent ({
...
+    // initialize your metrics collector ...
+    this.tid = setInterval(function () {
+      // get every 30 seconds some metrics 
+      // SPM gets an array of metrics for a specific app
+       agent.addMetrics({
+       	measurement: 'test-app', 
+       	tags: {role: 'frontend'}, 
+       	fields: {requestCount: 1}}
+       )
+    }, client.config.collectionInterval)
```
[the remark this answers is not in the document]
Comment (megastef) [reply README.md]: I will add 

```
// Typically agents collect metrics periodically, every N seconds. The time between 
// two collection activities is the collectionInterval, specified in milliseconds
```

File: README.md (hunk 4 above)
Comment (megastef) [reply README.md]: changed. 
```
// log collected metrics to console
// observe all metrics
testAgent.on('metrics', console.log)
// observe a specific metric using the measurment name
testAgent.on('process.memory', console.log)
```

**Pull request merged**